src/go/rpk/pkg/cli/registry/schema/create.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package schema
+
+import (
+	"os"
+	"strings"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/schemaregistry"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/twmb/franz-go/pkg/sr"
+)
+
+func newCreateCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var schemaFile string
+	cmd := &cobra.Command{
+		Use:   "create SUBJECT --schema {filename}",
+		Short: "Create a schema for the given subject",
+		Long: `Create a schema for the given subject.
+
+This uploads a schema to the registry, creating the schema if it does not
+exist. The schema type is detected by the filename extension: ".avro" for Avro
+and ".proto" for Protobuf. You can manually specify the type with the --type
```

Comment:
The --type flag does not exist?

Reply:
I'm going to delete this part of the doc, or do we want a `type` flag and avoid the file extension as a hard requirement?

Reply:
--type might actually be valuable so we don't require specific file extensions